diff --git a/src/vm-configurator.vala b/src/vm-configurator.vala
index ede5e0e..010faa0 100644
--- a/src/vm-configurator.vala
+++ b/src/vm-configurator.vala
@@ -443,11 +443,11 @@ private class Boxes.VMConfigurator {
 
     private static void set_video_config (Domain domain, InstallerMedia install_media) {
         var video = new DomainVideo ();
-        video.set_model (DomainVideoModel.QXL);
+        //video.set_model (DomainVideoModel.QXL);
 
-        if (install_media.supports_virtio_gpu) {
+        //if (install_media.supports_virtio_gpu) {
             video.set_model (DomainVideoModel.VIRTIO);
-        }
+        //}
 
         domain.add_device (video);
     }
