diff --git a/src/vm-configurator.vala b/src/vm-configurator.vala
index c75cd45..521ab0c 100644
--- a/src/vm-configurator.vala
+++ b/src/vm-configurator.vala
@@ -461,11 +461,11 @@ private static void set_os_config (Domain domain, InstallerMedia install_media,
 
     private static void set_video_config (Domain domain, InstallerMedia install_media) {
         var video = new DomainVideo ();
-        video.set_model (DomainVideoModel.QXL);
+        //video.set_model (DomainVideoModel.QXL);
 
-        if (install_media.supports_virtio_gpu) {
+        //if (install_media.supports_virtio_gpu) {
             video.set_model (DomainVideoModel.VIRTIO);
-        }
+        //}
 
         domain.add_device (video);
     }
