diff --git a/src/vm-configurator.vala b/src/vm-configurator.vala
index 521ab0c..bef9f95 100644
--- a/src/vm-configurator.vala
+++ b/src/vm-configurator.vala
@@ -103,8 +103,10 @@ public static Domain create_domain_config (InstallerMedia install_media, string
 
         add_usb_support (domain, install_media);
 
+/*
         if (!App.is_running_in_flatpak ())
             add_smartcard_support (domain);
+*/
 
         set_video_config (domain, install_media);
         set_sound_config (domain, install_media);
@@ -670,12 +672,14 @@ else if (install_media is InstalledMedia)
         }
     }
 
+/*
     public static void add_smartcard_support (Domain domain) {
         var smartcard = new DomainSmartcardPassthrough ();
         var vmc = new DomainChardevSourceSpiceVmc ();
         smartcard.set_source (vmc);
         domain.add_device (smartcard);
     }
+*/
 
     public static void add_usb_support (Domain domain, InstallerMedia install_media) {
 
