diff --git a/src/libvirt-machine.vala b/src/libvirt-machine.vala
index 054f722..fc90003 100644
--- a/src/libvirt-machine.vala
+++ b/src/libvirt-machine.vala
@@ -63,7 +63,7 @@
 #if HAS_SPICE
             GLib.List<GVirConfig.DomainDevice> devices = null;
             foreach (var device in domain_config.get_devices ()) {
-                if (device is GVirConfig.DomainGraphicsSpice) {
+                if (device is GVirConfig.DomainGraphicsVnc) {
                     var graphics_device = VMConfigurator.create_graphics_device (_acceleration_3d);
 
                     devices.prepend (graphics_device);
diff --git a/src/vm-configurator.vala b/src/vm-configurator.vala
index bef9f95..26a7199 100644
--- a/src/vm-configurator.vala
+++ b/src/vm-configurator.vala
@@ -742,12 +742,9 @@ private static DomainChannel create_spice_agent_channel () {
         return channel;
     }
 
-    public static DomainGraphicsSpice create_graphics_device (bool accel3d = false) {
-        var graphics = new DomainGraphicsSpice ();
-        graphics.set_autoport (false);
-        graphics.set_gl (accel3d);
-        graphics.set_image_compression (DomainGraphicsSpiceImageCompression.OFF);
-
+    public static DomainGraphicsVnc create_graphics_device (bool accel3d = false) {
+        var graphics = new DomainGraphicsVnc ();
+        graphics.set_autoport (true);
         return graphics;
     }
 
